WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.kt

```diff
@@ -72,37 +72,37 @@ abstract class BlockProcessor internal constructor(@JvmField var localId: String
      * @return A string containing content with ids and urls replaced
      */
     @JvmOverloads
-    fun processBlock(block: String, isSelfClosingTag: Boolean = false) = if (splitBlock(block, isSelfClosingTag)) {
-        if (processBlockJsonAttributes(jsonAttributes)) {
-            if (isSelfClosingTag) {
-                // return injected block
-                StringBuilder()
-                    .append("<!-- wp:")
-                    .append(blockName)
-                    .append(" ")
-                    .append(jsonAttributes) // json parser output
-                    .append(" /-->")
-                    .toString()
-            } else if (processBlockContentDocument(blockContentDocument)) {
-                // return injected block
-                StringBuilder()
-                    .append("<!-- wp:")
-                    .append(blockName)
-                    .append(" ")
-                    .append(jsonAttributes) // json parser output
-                    .append(" -->\n")
-                    .append(blockContentDocument?.body()?.html()) // HTML parser output
-                    .append(closingComment)
-                    .toString()
-            } else {
-                block
-            }
-        } else {
-            processInnerBlock(block) // delegate to inner blocks if needed
+    fun processBlock(block: String, isSelfClosingTag: Boolean = false) = when {
+        !splitBlock(block, isSelfClosingTag) -> block // leave block unchanged
+        jsonAttributes?.let { !processBlockJsonAttributes(it) } == true -> {
+            // delegate to inner blocks if needed
+            processInnerBlock(block)
+        }
+        isSelfClosingTag -> {
+            // return injected block
+            StringBuilder()
+                .append("<!-- wp:")
+                .append(blockName)
+                .append(" ")
+                .append(jsonAttributes) // json parser output
+                .append(" /-->")
+                .toString()
         }
-    } else {
-        // leave block unchanged
-        block
+
+        blockContentDocument?.let { processBlockContentDocument(it) } == true -> {
+            // return injected block
+            StringBuilder()
+                .append("<!-- wp:")
+                .append(blockName)
+                .append(" ")
+                .append(jsonAttributes) // json parser output
+                .append(" -->\n")
+                .append(blockContentDocument?.body()?.html()) // HTML parser output
+                .append(closingComment)
+                .toString()
+        }
+
+        else -> block // leave block unchanged
```

Comment:
Please review this `processBlock()` function more carefully to ensure that I didn't change any behavior.

Reply:
Won't lie. This was hard to follow 😅 . The function appears same as before though 👍🏻